### ISSUE TITLE
CDX/SPDX SBOM support where local image is used without registry or repo

### ISSUE THREAD

**Issue by pxp928** [guacsec/guac#387]
```
"metadata": {
    "timestamp": "2026-03-15T12:33:32Z",
    "tools": [
      {
        "vendor": "anchore",
        "name": "syft",
        "version": "0.65.0-SNAPSHOT-e480443"
      }
    ],
    "component": {
      "bom-ref": "r52dj2j5j2jj1",
      "type": "container",
      "name": "dev-test-container"
    }
  },
  ```
  This aligns with (CDX Spec)[https://cyclonedx.org/docs/1.4/json/#components_items_name] where the name is shortened.
  
  Update SPDX to also support a local container and match purl spec.

**Comment by pxp928** [comment]
Long term for the Beta release is the move away from using heuristics as it's too error-prone. This was merged into [heuristic-dev](https://github.com/guacsec/guac/tree/heuristic-dev) branch for the time being to unblock users.

**Comment by pxp928** [comment]
Fixed in the current version of the CDX parser. Updated to handle various name lengths:

```
	switch len(splitImage) {
	case 3:
		repositoryURL = splitImage[0] + "/" + splitImage[1] + "/" + splitTag[0]
	case 2:
		repositoryURL = splitImage[0] + "/" + splitTag[0]
	case 1:
		repositoryURL = splitImage[0]
	default:
		repositoryURL = ""
	}
```
